core/transaction/transaction.go

```diff
@@ -0,0 +1,39 @@
+package transaction
+
+import (
+	"time"
+)
+
+// Transaction is a single charging transaction with status and reservation and payment data
+type Transaction struct {
+	ID          uint `gorm:"primarykey"`
+	LoadpointId int
+	Loadpoint   string
+	VehicleId   int
```

Comment:
Vehicle ID == Index in der Konfiguration?

Das ist kein stabiler Identifier für Langzeitdaten. Fahrzeuge können ja kommen und gehen.

Auch wenn ich das bislang eigentlich immer ausgeschlossen habe. Vielleicht machts doch Sinn mal drüber nach zu denken. Wir könnten die DB auch für die Konfiguration nutzen (evcc.yaml ablösen). Das würde einiges vereinfachen (programmatische Änderung der Config, saubere Primärschlüssel). Dann wäre es bspw. einfacher ein Fahrzeug zu "archivieren". Es könnte dann trotzdem in der DB bleiben sofern es Referenzen darauf gibt (Ladelog). Mit der yaml ginge das theoretisch natürlich auch, da haben wir aber keine Sicherheit weil an der händisch geschraubt wird.

Reply:
> Das ist kein stabiler Identifier für Langzeitdaten. 

Besser Vehicle, also `vehicle.Title()`.